crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -170,6 +170,17 @@ impl Context {
             // Note: that this produces unnecessary constraints when
             // this Eq instruction is being used for a constrain statement
             BinaryOp::Eq => self.acir_context.eq_var(lhs, rhs),
+            // TODO: This is going to be somewhat inefficient initially since
```

Comment:
This is not a blocker, but we should open up an issue for this. 


It adds an extra constraint per MoreThanEq done in SSA IR, but there are a lot of ways to mitigate this, so low priority

Reply:
https://github.com/noir-lang/noir/issues/1444